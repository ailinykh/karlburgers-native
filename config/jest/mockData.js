export const products = [
  {
    id: "396beeeb-6ace-4d34-b006-3bd22648005c",
    name: "Карл-Бургер",
    code: "00074",
    description: "Фирменный бургер с говяжьей котлетой, беконом и луковым джемом.\n<br>\n<br>\n<br>\n<b>Снаружи:</b> Булочка с кунжутом.\n<br>\n<b>Внутри:</b> Говяжья котлета, бекон, сыр Моцарелла, листья салата, соленый огурец, чесночный соус, луковый джем и соус барбекю.\n<br>\n<br>\n<img src=\"http://www.karlburgers.ru/wp-content/uploads/2016/03/imidj_karl.jpg\"> \n",
    order: 2,
    parentGroup: "99dd07e2-fc7f-4a30-8d4b-8a8528272729",
    imageId: "58980899-e619-4a33-a3db-2891f84e7241",
    price: 270,
  },
  {
    id: "5e65b415-7c8d-4832-8f56-dcd97e181c57",
    name: "Классика жанра",
    code: "00083",
    description: "Чизбургер. Говяжья котлета, сыр и овощи.\n<br>\n<br>\n<br>\n<br>\n<b>Снаружи:</b> Булочка с кунжутом.\n<br>\n<b>Внутри:</b> Говяжья котлета, сыр Чеддер, листья салата, соленый огурец, чесночный соус, кетчуп и красный лук.\n\n<br>\n<br>\n<img src=\"http://www.karlburgers.ru/wp-content/uploads/2016/03/i_chiz.jpg\">",
    order: 3,
    parentGroup: "99dd07e2-fc7f-4a30-8d4b-8a8528272729",
    imageId: "530fb33d-62d5-430f-9aea-90f839fd7cfb",
    price: 250,
  },
  {
    id: "b5bcdca1-4d2b-4a9d-a40f-6bf61fda038b",
    name: "Море-море",
    code: "00072",
    description: "Бургер с филе трески и свежими овощами.\n<br>\n<br>\n<br>\n<br>\n<b>Снаружи:</b> Булочка с кунжутом.\n<br>\n<b>Внутри:</b> Филе трески, свежий огурец, листья салата, соус терияки и луковая крошка.\n<br>\n<br>\n<img src=\"http://www.karlburgers.ru/wp-content/uploads/2016/03/i_more.jpg\"> \n",
    order: 4,
    parentGroup: "99dd07e2-fc7f-4a30-8d4b-8a8528272729",
    imageId: "c39f8ba7-42f8-4c55-a03b-de44cce562fc",
    price: 250,
  },
];
