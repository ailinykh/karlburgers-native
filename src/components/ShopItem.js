import React, { Component } from 'react';
import { StyleSheet, View, Image, Text, Dimensions, TouchableHighlight, TouchableOpacity } from 'react-native';
import { Actions } from 'react-native-router-flux';
import Icon from 'react-native-vector-icons/Ionicons';
import { IIKO_RESTARAUNT_ID, KB_ORANGE } from '../constants';

export default class ShopItem extends Component {
  render() {
    return (
      <TouchableHighlight onPress={() => Actions.product({product: this.props.product, title: this.props.product.name})} style={styles.item} underlayColor='whitesmoke'>
        <View>
          <Image style={styles.image}
            resizeMode={Image.resizeMode.contain}
            defaultSource={require('../images/kb-placeholder.png')}
            source={{uri: `http://deliverywiget.iiko.ru/Content/User/${IIKO_RESTARAUNT_ID}/${this.props.product.imageId}_Medium_.png`}}
          />
          <View style={styles.orderContainer}>
            <Text style={styles.priceContainer}>
              <Text style={styles.price}>{this.props.product.price}</Text> руб.
            </Text>
            <TouchableOpacity onPress={() => this.props.addToCartAction(this.props.product)}>
              <Text style={styles.cartContainer}>
                <Icon name="ios-cart-outline" size={19} style={styles.cartButton}/> {this.props.cartCount}
              </Text>
            </TouchableOpacity>
          </View>
          <Text style={styles.title}>
            {this.props.product.name}
          </Text>
          <Text style={styles.description}>
            {this.props.product.description.replace(/<br>/, '\n').split('\n')[0]}
          </Text>
        </View>
      </TouchableHighlight>
    );
  }
}

var width = Dimensions.get('window').width;

export const styles = StyleSheet.create({
  item: {
    // backgroundColor: '#CCC',
    padding: 10,
    width: width*.5,
    height: width > 320 ? width*.87 : width*.94,
    borderBottomWidth: 1,
    borderBottomColor: '#ddd',
  },
  image: {
    height: width*.4
  },
  orderContainer: {
    padding: width > 320 ? 10 : 5,
    flexDirection: 'row',
    justifyContent: 'space-between',
    // backgroundColor: 'aqua'
  },
  priceContainer: {
    fontSize: 17,
    // backgroundColor: 'burlywood',
  },
  cartContainer: {
    paddingLeft: 10,
    paddingRight: 10,
    textAlign: 'right',
    color: KB_ORANGE,
    fontSize: 19,
    // backgroundColor: 'beige',
  },
  cartButton: {

  },
  price: {
    fontWeight:'bold',
  },
  title: {
    paddingBottom: 5,
    fontSize: 20,
    textAlign: 'center'
  },
  description: {
    textAlign: 'center'
  }
});
