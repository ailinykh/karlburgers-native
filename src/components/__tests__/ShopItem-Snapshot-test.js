import React from 'react';
import renderer from 'react-test-renderer';

import ShopItem from '../ShopItem';
import { products } from '../../../config/jest/mockData';

it('renders a ShopItem using Snapshots', () => {
  expect(renderer.create(
    <ShopItem
      product={products[0]}
      cartCount={0}
      addToCartAction={jest.fn()}
    />
  )).toMatchSnapshot();

  const component = renderer.create(
    <ShopItem
      product={products[0]}
      cartCount={0}
      addToCartAction={jest.fn()}
    />
  );
});
