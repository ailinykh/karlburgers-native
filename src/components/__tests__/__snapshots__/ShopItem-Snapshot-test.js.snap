exports[`test renders a ShopItem using Snapshots 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "borderBottomColor": "#ddd",
        "borderBottomWidth": 1,
        "height": 652.5,
        "padding": 10,
        "width": 375,
      },
    ]
  }
  testID={undefined}>
  <View>
    <Image
      defaultSource={1}
      resizeMode="contain"
      source={
        Object {
          "uri": "http://deliverywiget.iiko.ru/Content/User/5c77b3eb-0b8b-11e6-80c8-d8d385655247/58980899-e619-4a33-a3db-2891f84e7241_Medium_.png",
        }
      }
      style={
        Object {
          "height": 300,
        }
      } />
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 17,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontWeight": "bold",
            }
          }>
          270
        </Text>
         руб.
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#f8863d",
              "fontSize": 19,
              "paddingLeft": 10,
              "paddingRight": 10,
              "textAlign": "right",
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 19,
                },
                Object {},
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }>
            
          </Text>
           
          0
        </Text>
      </View>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 20,
          "paddingBottom": 5,
          "textAlign": "center",
        }
      }>
      Карл-Бургер
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "textAlign": "center",
        }
      }>
      Фирменный бургер с говяжьей котлетой, беконом и луковым джемом.
    </Text>
  </View>
</View>
`;
